test/ret_test.exs

```diff
@@ -29,307 +30,201 @@ defmodule RetTest do
-      })
-
-      Repo.insert(%HubInvite{
-        hub: hub,
-        hub_invite_sid: "fake-invite-sid"
-      })
-
-      Repo.insert(%HubRoleMembership{
-        hub: hub,
-        account: test_hub_member_account
-      })
-
-      Repo.insert(%RoomObject{
-        hub: hub,
-        account: test_account,
-        object_id: "fake object id",
-        gltf_node: "fake gltf node"
-      })
-
-      Repo.insert(%WebPushSubscription{
         hub: hub,
-        endpoint: "fake-endpoint",
-        p256dh: "fake-key",
-        auth: "fake-auth-key"
+        type: :discord
       })
 
-      1 = count(Hub, test_account)
-      1 = count(AccountFavorite, hub)
-      1 = count(HubBinding, hub)
-      1 = count(HubInvite, hub)
-      1 = count(HubRoleMembership, hub)
-      1 = count(RoomObject, hub)
-      1 = count(WebPushSubscription, hub)
-
-      assert :ok = Ret.delete_account(admin_account, test_account)
-
-      assert 0 === count(Hub, test_account)
-      assert 0 === count(AccountFavorite, hub)
-      assert 0 === count(HubBinding, hub)
-      assert 0 === count(HubInvite, hub)
-      assert 0 === count(HubRoleMembership, hub)
-      assert 0 === count(RoomObject, hub)
-      assert 0 === count(WebPushSubscription, hub)
+      Repo.insert(%AccountFavorite{account: test_account, hub: hub})
+      Repo.insert(%HubInvite{hub: hub, hub_invite_sid: "fake-invite-sid"})
+      Repo.insert(%HubRoleMembership{account: test_hub_member_account, hub: hub})
+
+      Repo.insert(%RoomObject{account: test_account, gltf_node: "fake gltf node", hub: hub, object_id: "fake object id"})
+
+      Repo.insert(%WebPushSubscription{auth: "fake-auth-key", endpoint: "fake-endpoint", hub: hub, p256dh: "fake-key"})
+      1 = count_where(Hub, created_by_account_id: test_account.account_id)
+      1 = count_where(AccountFavorite, hub_id: hub.hub_id)
+      1 = count_where(HubBinding, hub_id: hub.hub_id)
+      1 = count_where(HubInvite, hub_id: hub.hub_id)
+      1 = count_where(HubRoleMembership, hub_id: hub.hub_id)
+      1 = count_where(RoomObject, hub_id: hub.hub_id)
+      1 = count_where(WebPushSubscription, hub_id: hub.hub_id)
+
+      assert :ok === Ret.delete_account(test_account.account_id, admin_account)
+      assert 0 === count_where(Hub, created_by_account_id: test_account.account_id)
+      assert 0 === count_where(AccountFavorite, hub_id: hub.hub_id)
+      assert 0 === count_where(HubBinding, hub_id: hub.hub_id)
+      assert 0 === count_where(HubInvite, hub_id: hub.hub_id)
+      assert 0 === count_where(HubRoleMembership, hub_id: hub.hub_id)
+      assert 0 === count_where(RoomObject, hub_id: hub.hub_id)
+      assert 0 === count_where(WebPushSubscription, hub_id: hub.hub_id)
     end
 
     test "deletes entities associated with an account, even when they belong to a hub owned by another account" do
       {:ok, admin_account: admin_account} = create_admin_account("test_admin")
       hub_owner = create_account("test_owner")
```

Comment:
It is unfortunate that what we call _rooms_, Reticulum calls _hubs_, especially now that `Dash.Hub` exists for the managed product and a `Dash.Hub` is what we mean when we say "a hub".

Reply:
There are some good refactoring opportunities. I’d love to do some [event storming](https://www.eventstorming.com/) and [domain modeling](https://en.wikipedia.org/wiki/Domain-driven_design) and work towards a [ubiquitous language](https://martinfowler.com/bliki/UbiquitousLanguage.html).